spec/src/modules/tracker.js

```diff
@@ -1898,6 +1898,60 @@ describe(`ConstructorIO - Tracker${bundledDescriptionSuffix}`, () => {
 
       expect(tracker.trackSearchSubmit(term, requiredParameters)).to.equal(true);
     });
+
+    it('Should properly obfuscate PII in a path parameter', (done) => {
+      const termWithPII = ' test-email@gmail.com';
+      const replaceBy = '<email_omitted>';
+      const { tracker } = new ConstructorIO({
+        apiKey: testApiKey,
+        fetch: fetchSpy,
+        ...requestQueueOptions,
+      });
+
+      tracker.on('success', (responseParams) => {
+        const requestUrl = fetchSpy.args[0][0];
+
+        // Request
+        expect(fetchSpy).to.have.been.called;
+        expect(requestUrl).to.include(replaceBy);
+        expect(requestUrl).to.not.include(termWithPII);
+
+        // Response
+        expect(responseParams).to.have.property('method').to.equal('GET');
+        expect(responseParams).to.have.property('message').to.equal('ok');
+
+        done();
+      });
+
+      expect(tracker.trackSearchSubmit(termWithPII, { originalQuery: termWithPII })).to.equal(true);
+    });
+
+    it('Should properly obfuscate PII in a query parameter', (done) => {
+      const termWithPII = ' test-email@gmail.com';
+      const replaceBy = '<email_omitted>';
```

Comment:
What do you think about `replaceWith` instead of `replaceBy`? Not at all important though - feel free to ignore.

Reply:
I like it. I prefer `replaceWith` instead of `replaceBy`. Will push a change shortly to update :)

Reply:
done: [rename replaceBy to replaceWith](https://github.com/Constructor-io/constructorio-client-javascript/pull/317/commits/b8c2c1c367025d34047e7e8c877ee0d9a79ac3dc)

Reply:
and: [replace replaceBy with replaceWith](https://github.com/Constructor-io/constructorio-client-javascript/pull/317/commits/3d56b2d3201deacf6d17becf5d9f760b2fa0f076)